mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -186,34 +188,55 @@ def get_model_endpoint_real_time_metrics(
     def get_records(
         self,
         table: str,
-        start: str,
-        end: str,
+        start: datetime,
+        end: datetime,
         columns: typing.Optional[list[str]] = None,
         filter_query: str = "",
+        interval: str = "",
+        agg: typing.Optional[list] = None,
+        limit: int = 0,
+        sliding_window: str = "",
         timestamp_column: str = mm_schemas.EventFieldType.TIME,
     ) -> pd.DataFrame:
         """
         Getting records from TSDB data collection.
         :param table:            Either a supertable or a subtable name.
-        :param columns:          Columns to include in the result.
-        :param filter_query:     Optional filter expression as a string. The filter structure depends on the TSDB
-                                 connector type.
         :param start:            The start time of the metrics.
         :param end:              The end time of the metrics.
+        :param columns:          Columns to include in the result.
+        :param filter_query:     Optional filter expression as a string. TDengine supports SQL-like syntax.
+        :param interval:         The interval to aggregate the data by. Note that if interval is provided,
+                                 agg must bg provided as well. Provided as a string in the format of
+                                 '1m', '1h', etc.
+        :param agg:              The aggregation functions to apply on the columns. Note that if agg is provided,
+                                 interval must bg provided as well. Provided as a list of strings in the format of
+                                 ['sum', 'avg', 'count', ...].
+        :param limit:            The maximum number of records to return.
+        :param sliding_window:   The time step for which the time window moves forward. Note that if sliding_window is
+                                 provided, interval must be provided as well. Provided as a string in the format of
+                                 '1m', '1h', etc.
         :param timestamp_column: The column name that holds the timestamp.
 
         :return: DataFrame with the provided attributes from the data collection.
         :raise:  MLRunInvalidArgumentError if query the provided table failed.
         """
-
-        filter_query += f" project = '{self.project}'"
+        with StringIO() as query:
+            if filter_query:
+                query.write(filter_query)
+                query.write(" and ")
+            query.write(f"project = '{self.project}'")
+            filter_query = query.getvalue()
```

Comment:
Overkill to use `StringIO` here, this is only a few short strings being concatenated. Can be something like:
```suggestions
        project_condition = f"project = '{self.project}'"
        filter_query = (
            f"{filter_query} and {project_condition}"
            if filter_query
            else project_condition
        )
```